### PR TITLE
Fix DOMStringMap's stringifier behaviour according to the spec

### PR DESCRIPTION
**Pull request by nox**
It should just return its associated attribute's value, if any.

https://github.com/whatwg/dom/issues/105

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/8959)

### CODE REVIEW

**Comment by nox** [comment]
Partially fixes #8511.

**Comment by KiChjang** [comment]
```
Ran 7097 tests finished in 884.0 seconds.
  • 7096 ran as expected. 8 tests skipped.
  • 1 tests crashed unexpectedly

Tests with unexpected results:
  ▶ CRASH [expected PASS] /css21_dev/html4/margin-bottom-064.htm
  │ 
  │ was ready to save, resetting ready_to_save_state
  │ Shutting down the Constellation after generating an output file or exit flag specified
  │ thread 'ScriptTask PipelineId { namespace_id: PipelineNamespaceId(0), index: PipelineIndex(0) }' panicked at 'called `Result::unwrap()` on an `Err` value: Error { repr: Custom(Custom { kind: Other, error: StringError("recursive IPC channel use during serialization") }) }', ../src/libcore/result.rs:741
  │ stack backtrace:
  │    1:        0x10d120ab8 - sys::backtrace::tracing::imp::write::hca66179add43b2e5lLt
  │    2:        0x10d122b7f - panicking::log_panic::_<closure>::closure.40842
  │    3:        0x10d122622 - panicking::log_panic::h7a04edac7e1f2797HEx
  │    4:        0x10d10e4c6 - sys_common::unwind::begin_unwind_inner::h39a94087039ae0efnOs
  │    5:        0x10d10e8ae - sys_common::unwind::begin_unwind_fmt::h7be0824d83765cd5tNs
  │    6:        0x10d1200d7 - rust_begin_unwind
  │    7:        0x10d145160 - panicking::panic_fmt::h871c0ef100fda9e9LFK
  │    8:        0x10c77be87 - mem::_<impl>::send::haf009788f0ee7115Bea
  │    9:        0x10c330640 - sys_common::unwind::try::try_fn::try_fn::h17387133094171496326
  │   10:        0x10d11fef8 - __rust_try
  │   11:        0x10d11cffe - sys_common::unwind::try::inner_try::h819d2da8e3dc4516VKs
  │   12:        0x10c331811 - boxed::_<impl>::call_box::call_box::h2443647416074846130
  │   13:        0x10d121e0d - sys::thread::_<impl>::new::thread_start::h328957c252b95d3cOYw
  │   14:     0x7fff8416a059 - _pthread_body
  │   15:     0x7fff84169fd6 - _pthread_start
  │ thread 'ScriptTask PipelineId { namespace_id: PipelineNamespaceId(0), index: PipelineIndex(0) }' panicked at 'called `Option::unwrap()` on a `None` value', ../src/libcore/option.rs:367
  │ stack backtrace:
  │    1:        0x10d120ab8 - sys::backtrace::tracing::imp::write::hca66179add43b2e5lLt
  │    2:        0x10d122b7f - panicking::log_panic::_<closure>::closure.40842
  │    3:        0x10d122622 - panicking::log_panic::h7a04edac7e1f2797HEx
  │    4:        0x10d10e3c4 - sys_common::unwind::begin_unwind_inner::h39a94087039ae0efnOs
  │    5:        0x10d10e8ae - sys_common::unwind::begin_unwind_fmt::h7be0824d83765cd5tNs
  │    6:        0x10d1200d7 - rust_begin_unwind
  │    7:        0x10d145160 - panicking::panic_fmt::h871c0ef100fda9e9LFK
  │    8:        0x10d14545c - panicking::panic::hbe2e3a5159c3d178iEK
  │    9:        0x10bee69d9 - page::_<impl>::window::he9494a240cfd38dfqqZ
  │   10:        0x10c32d714 - script_task::_<impl>::drop::h2beced7d2c679ffeiD0
  │   11:        0x10c331160 - sys_common::unwind::try::try_fn::try_fn::h17387133094171496326
  │   12:        0x10d11fef8 - __rust_try
  │   13:        0x10d11cffe - sys_common::unwind::try::inner_try::h819d2da8e3dc4516VKs
  │   14:        0x10c331811 - boxed::_<impl>::call_box::call_box::h2443647416074846130
  │   15:        0x10d121e0d - sys::thread::_<impl>::new::thread_start::h328957c252b95d3cOYw
  │   16:     0x7fff8416a059 - _pthread_body
  │   17:     0x7fff84169fd6 - _pthread_start
  └ thread panicked while panicking. aborting.
```

**Comment by nox** [comment]
That's #8945.

**Comment by jdm** [comment]
@bors-servo: retry
* #8961

**Comment by jdm** [comment]
```
  ▶ Unexpected subtest result in /dom/nodes/Element-classlist.html:
  │ FAIL [expected PASS] empty classList should return the empty string since the ordered set parser skip the whitespaces
  │   → assert_equals: implicit expected "" but got " "
  │ 
  │ @http://web-platform.test:8000/dom/nodes/Element-classlist.html:54:3
  │ Test.prototype.step@http://web-platform.test:8000/resources/testharness.js:1381:20
  │ test@http://web-platform.test:8000/resources/testharness.js:495:9
  └ @http://web-platform.test:8000/dom/nodes/Element-classlist.html:53:1
```

File: components/script/dom/domtokenlist.rs [hunk 1]
```diff
@@ -131,8 +131,7 @@ impl DOMTokenListMethods for DOMTokenList {
 
     // https://dom.spec.whatwg.org/#stringification-behavior
```
Comment (Ms2ger) [inline components/script/dom/domtokenlist.rs]: Fix to <https://dom.spec.whatwg.org/#dom-domtokenlist-stringifier> or <https://dom.spec.whatwg.org/#concept-dtl-serialize>, please. r+ with that.

Comment (nox) [inline components/script/dom/domtokenlist.rs]: Good catch.

**Pull request merged**